components/servo/tests/webview.rs

```diff
@@ -778,3 +730,57 @@ fn test_can_go_forward_and_can_go_back() {
     assert!(webview.can_go_forward());
     assert!(!webview.can_go_back());
 }
+
+/// A test that ensures that protocol handler registrations are send to the embedder
+#[test]
+fn test_protocol_handler_registration_register() {
+    let servo_test = ServoTest::new();
+    struct WebViewProtocolHandlerTestDelegate {
+        servo: Rc<Servo>,
+        protocol_handler_request: RefCell<Option<ProtocolHandlerRegistration>>,
+        registration_changed: Cell<bool>,
+    }
+
+    impl WebViewDelegate for WebViewProtocolHandlerTestDelegate {
+        fn request_protocol_handler(
+            &self,
+            _webview: WebView,
+            protocol_handler_registration: ProtocolHandlerRegistration,
+            allow_deny_request: AllowOrDenyRequest,
+        ) {
+            *self.protocol_handler_request.borrow_mut() = Some(protocol_handler_registration);
+            self.registration_changed.set(true);
+            allow_deny_request.allow();
+        }
+    }
+
+    let delegate = Rc::new(WebViewProtocolHandlerTestDelegate {
+        servo: servo_test.servo.clone(),
+        protocol_handler_request: Default::default(),
+        registration_changed: Default::default(),
+    });
+
+    let webview = WebViewBuilder::new(servo_test.servo())
+        .delegate(delegate.clone())
+        .url(
+            Url::parse(
+                "data:text/html,<!DOCTYPE html><script>\
+                    window.navigator.registerProtocolHandler('web+test', 'http://localhost:8000/web/test/');\
```

Comment:
The test here doesn't work, because `registerProtocolHandler` can only be called for the same origin. However, none of the webview tests load a URL that has an origin. They call use `data:text/html`.

How can I refer to local resources so that we spin up a webserver and run that?

All of this is to say, I think instead of having a unit test for this, I rather rely on the WPT tests for these: https://github.com/servo/servo/blob/main/tests/wpt/tests/html/webappapis/system-state-and-capabilities/the-navigator-object/protocol-handler-fragment-nosw.https.html